File: classes01/python/erode.py
import cv2

image=cv2.imread("../../data/images/truth.png",cv2.IMREAD_COLOR)

erosionSize = 6
element = cv2.getStructuringElement(cv2.MORPH_CROSS,(erosionSize*2+1,erosionSize*2+1),(erosionSize,erosionSize))
cv2.imshow("element",element)

imageEroded=cv2.erode(image,element)

cv2.imshow("origin",image)
cv2.imshow("Eroded Image",imageEroded)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: classes01/cpp/erode.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void)
{
    Mat image;
    image=imread("../../data/images/truth.png",IMREAD_COLOR);
    
    int erosionSize=6;
    Mat element=getStructuringElement(MORPH_CROSS,Size(2*erosionSize+1,2*erosionSize+1),Point(erosionSize,erosionSize));

    Mat imageEroded;
    erode(image,imageEroded,element);

    imshow("Original",image);
    imshow("Eroded Image",imageEroded);

    waitKey(0);
    destroyAllWindows();

    return EXIT_SUCCESS;
}